src/temaWeek8Exception/main/Repository.java

```diff
@@ -0,0 +1,101 @@
+package temaWeek8Exception.main;
+
+import java.time.Year;
+import java.util.*;
+
+public class Repository {
+	String firstName = "";
+	String lastName = "";
+	String dateOfBirth = "";
+	String gender = "";
+	String ID = "";
+	
+	
+	Student student = new Student(firstName, lastName, dateOfBirth, gender, ID);
+	List<Student> studentRepo = new ArrayList<>();
+	
+	public static void main(String[] args) {
+	
+	}
+	
+//	create student method
+	public Student createStudent(String firstName, String lastName, String dateOfBirth, String gender, String ID) {
+
+		firstName = student.addFirstName(firstName);//calling addFirstName method for validation of method argument
+		lastName = student.addLastName(lastName);//calling addLastName method for validation of method argument
+		dateOfBirth = student.addDateOfBirth(dateOfBirth);//calling addDateOfBirth method for validation of method argument
+		gender = student.addGender(gender);//calling addGender method for validation of method argument
+		ID = student.addID(ID);//calling addID method for validation of method argument
```

Comment:
Maybe it's easier to validate each field in the constructor so you don't have to call each method just for validation purposes. It's easy to miss one, especially if the model changes over time.